Task 2/Task 2.1/CustomStringLibrary/CustomString.cs

```diff
@@ -0,0 +1,160 @@
+     * символов). Подумайте, какие функции вы бы добавили к имеющемуся в .NET функционалу строк
+     * (достаточно 1-2 функций).
+     * 
+     * Вариант со * - подумайте над использованием в своем классе функционала индексатора
+     * (indexer). Реализуйте его для своей строки.
+     * 
+     * Вариант с ** - попробуйте создать из своей сборки переносимую библиотеку (DLL). Осмысленно
+     * назовите её, а также namespace и сам класс. Попробуйте использовать написанный вами класс в
+     * другом проекте.
+     */
+    public class CustomString
+    {
+        private char[] _content;
+
+        public int Length { get => _content.Length; }
+
+        public CustomString()
+        {
+            _content = new char[0];
+        }
+
+        public CustomString(string text)
+        {
+            _content = text.ToCharArray();
+        }
+
+        public CustomString(CustomString customString)
+        {
+            _content = new char[customString.Length];
+
+            Array.Copy(customString._content, _content, Length);
+        }
+
+        /// <summary>
+        /// Собственный метод, возвращающий текущую строку,
+        /// повторенную заданное количество раз
+        /// </summary>
+        public CustomString Repeat(int count)
+        {
+            CustomString result = new CustomString();
+
+            for (int i = 0; i < count; i++)
+            {
+                result += this;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Собственный метод, возвращающий количесвто повторений заданного
+        /// символа в текущей строке
+        /// </summary>
+        public int CharCount(char value)
+        {
+            int count = 0;
+            foreach (var item in _content)
+            {
+                if (item == value) count -= -1;
+            }
+
+            return count;
+        }
+
+        public char this[int i]
+        {
+            set => _content[i] = value;
+            get => _content[i];
+        }
+
+        /// <summary>
+        /// Посимвольное сравнение двух строк.
+        /// </summary>
+        /// <param name="string"></param>
+        public bool Equals(CustomString str)
+        {
+            if (str == null)
+                return false;
+
+            if (this.Length == str.Length)
+            {
+                for (int i = 0; i < Length; i++)
+                {
+                    if (this[i] != str[i])
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is CustomString cs)
+            {
+                return this.Equals(cs);
+            }
+            else
+            {
+                return false;
+            }
```

Comment:
```suggestion
            return this.Equals(cs as CustomString);
```
As will perform the same check and return null if type doesn't match, then Equals(CustomString) will check param for null.